src/i2c_builder.rs

```diff
@@ -0,0 +1,38 @@
+//! I2C interface factory
+
+/// Builder struct for an I2C interface. Driver options and interface are set using its methods.
+#[derive(Copy, Clone, Debug)]
+pub struct I2CDIBuilder {
+    i2c_addr: u8,
+}
+
+impl Default for I2CDIBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl I2CDIBuilder {
```

Comment:
To be fair, if you're transitioning away from the main builders, maybe this could get some love as well. This struct is just 2 constructors - one with a default address and one with a specific one.

Reply:
Good shout. I've changed it to a struct with static methods. There's an argument to make them free functions in a module, but it should be better now either way.